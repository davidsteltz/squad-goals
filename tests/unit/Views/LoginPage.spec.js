import { shallowMount } from '@vue/test-utils';
import Login from '@/views/LoginPage.vue';

describe('LoginPage', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Login);
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  test('should render signIn when it is selected', () => {
    const wrapper = shallowMount(Login, {
      data: () => { return { hasAccount: true }; }
    });

    expect(wrapper.findAll({ name: 'SignInComponent' }).length).toBe(1);
    expect(wrapper.findAll({ name: 'SignUpComponent' }).length).toBe(0);
  });

  test('should render signUp when it is selected', () => {
    const wrapper = shallowMount(Login, {
      data: () => { return { hasAccount: false }; }
    });

    expect(wrapper.findAll({ name: 'SignInComponent' }).length).toBe(0);
    expect(wrapper.findAll({ name: 'SignUpComponent' }).length).toBe(1);
  });

  test('renders correctly', () => {
    // shallowMount because this is a large view and we don't want to initialize all of its components
    // Instead we just check that they are present
    const fullRender = shallowMount(Login);
    expect(fullRender.element).toMatchSnapshot();
  });
});
