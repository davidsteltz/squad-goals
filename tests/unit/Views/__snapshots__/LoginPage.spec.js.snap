// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<div
  class="auth-page"
>
  <signin-stub
    togglesignup="function () { [native code] }"
  />
   
  <!---->
</div>
`;
